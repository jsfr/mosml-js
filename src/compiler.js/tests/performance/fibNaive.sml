prim_val ctimebegin_ : string -> unit = 1 "console.time";
prim_val ctimeend_ : string -> unit = 1 "console.timeEnd";
fun ctimeBegin n = ctimebegin_ n;
fun ctimeEnd n = ctimeend_ n;

fun repeat n f x =
    let val n = ref n
    in  while !n > 0 do (n := !n-1; f x) end;
fun timerep n f s = repeat n (fn x => (ctimeBegin(s);f x; ctimeEnd(s)));

fun naiveFib 0.0 = 1.0
  | naiveFib 1.0 = 1.0
  | naiveFib n = naiveFib (n-1.0) + naiveFib (n-2.0)

val _ = timerep 100 naiveFib "naiveFib" 35.0