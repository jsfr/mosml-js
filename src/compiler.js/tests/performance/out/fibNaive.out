==============================LAMBDA BEGIN==============================
(prim (set_global fibNaive.ctimeBegin/1) (fn (prim (ccall1 console.time) var:0)))
(prim (set_global fibNaive.ctimeEnd/2) (fn (prim (ccall1 console.timeEnd) var:0)))
(prim (set_global fibNaive.repeat/3) (fn (fn (fn let (prim (makeblock 250:1) var:2) in while (prim (test:gt) (prim (field 0) var:0) 0) do ((prim (setfield 0) var:0 (prim (smlsubint) (prim (field 0) var:0) 1)); (app var:2 var:1)) end))))
(prim (set_global fibNaive.timerep/4) (fn (fn (fn (app (prim (get_global fibNaive.repeat/3) ) var:2 (fn ((app (prim (get_global fibNaive.ctimeBegin/1) ) var:1); ((app var:2 var:0); (app (prim (get_global fibNaive.ctimeEnd/2) ) var:1)))))))))
(prim (set_global fibNaive.naiveFib/5) (fn ((case var:0 of 0.0 : 1.0 1.0 : 1.0) statichandle (prim (floatprim smladdfloat) (app (prim (get_global fibNaive.naiveFib/5) ) (prim (floatprim smlsubfloat) var:0 1.0)) (app (prim (get_global fibNaive.naiveFib/5) ) (prim (floatprim smlsubfloat) var:0 2.0))))))
let (app (prim (get_global fibNaive.timerep/4) ) 100 (prim (get_global fibNaive.naiveFib/5) ) "naiveFib" 35.0) in unspec end
==============================LAMBDA END================================
beautified fibNaive.js
==============================JS BEGIN==============================
naiveFib: 862ms
naiveFib: 904ms
naiveFib: 907ms
naiveFib: 862ms
naiveFib: 930ms
naiveFib: 870ms
naiveFib: 903ms
naiveFib: 865ms
naiveFib: 925ms
naiveFib: 858ms
naiveFib: 863ms
naiveFib: 889ms
naiveFib: 861ms
naiveFib: 921ms
naiveFib: 860ms
naiveFib: 898ms
naiveFib: 863ms
naiveFib: 921ms
naiveFib: 855ms
naiveFib: 863ms
naiveFib: 889ms
naiveFib: 861ms
naiveFib: 888ms
naiveFib: 862ms
naiveFib: 992ms
naiveFib: 862ms
naiveFib: 888ms
naiveFib: 866ms
naiveFib: 861ms
naiveFib: 925ms
naiveFib: 861ms
naiveFib: 899ms
naiveFib: 861ms
naiveFib: 917ms
naiveFib: 861ms
naiveFib: 888ms
naiveFib: 873ms
naiveFib: 862ms
naiveFib: 906ms
naiveFib: 860ms
naiveFib: 891ms
naiveFib: 861ms
naiveFib: 883ms
naiveFib: 845ms
naiveFib: 851ms
naiveFib: 885ms
naiveFib: 859ms
naiveFib: 886ms
naiveFib: 862ms
naiveFib: 909ms
naiveFib: 861ms
naiveFib: 877ms
naiveFib: 859ms
naiveFib: 844ms
naiveFib: 889ms
naiveFib: 861ms
naiveFib: 891ms
naiveFib: 861ms
naiveFib: 903ms
naiveFib: 861ms
naiveFib: 868ms
naiveFib: 885ms
naiveFib: 862ms
naiveFib: 925ms
naiveFib: 858ms
naiveFib: 898ms
naiveFib: 861ms
naiveFib: 918ms
naiveFib: 872ms
naiveFib: 861ms
naiveFib: 903ms
naiveFib: 859ms
naiveFib: 890ms
naiveFib: 858ms
naiveFib: 893ms
naiveFib: 860ms
naiveFib: 869ms
naiveFib: 880ms
naiveFib: 858ms
naiveFib: 908ms
naiveFib: 858ms
naiveFib: 925ms
naiveFib: 856ms
naiveFib: 884ms
naiveFib: 858ms
naiveFib: 866ms
naiveFib: 891ms
naiveFib: 858ms
naiveFib: 891ms
naiveFib: 858ms
naiveFib: 893ms
naiveFib: 859ms
naiveFib: 931ms
naiveFib: 901ms
naiveFib: 859ms
naiveFib: 898ms
naiveFib: 862ms
naiveFib: 919ms
naiveFib: 858ms
naiveFib: 938ms
==============================JS END================================
